docs/development/external-name-handling.md

```diff
@@ -0,0 +1,128 @@
+
+#### Delete()
+If our external resource has a field (written into status by Observe)  indicating it is in deletion state, we return.
+
+We use external-name to perform DELETE request.
+
+If the response is 404 – not found (or the equivalent API response for this API), we do NOT treat this as an error. This only happens when the resource was externally removed in the meantime since after Observe(), Delete() is not called if the resource does not exist. Technically we have a drift in here but it can be safely ignored since the drift already covers the desired state.
+
+If our delete request is successful, we return EVEN IF the deletion operation is performed async by the external system (in the next reconcile Observe(), the resourceExist field will be checked by crossplane-runtime and Delete() is called again if the resource still exists). Should the resource still exist in next reconcile, Delete() is triggered again but we check for the deletion state in the external system (= status.observation being something like `DELETING` or so) so we don’t perform another Delete(). If this is not possible, we have to perform another DELETE API call, treat a possible error response as error and return it. Once the resource is deleted in the external system, the Observe() function will determine resourceExist: false and crossplane-runtime removed the MR.
+
+Else, if we have an error in our request, return with error.
+
+### Importing of existing external resources
+For importing, setting the right external-name will match the external resource no matter the values in `spec.ForProvider`.
+
+If we set the `spec.forProvider` as required, the user can not create managed resources without those values and therefore must also set the required fields. The official import guide [3] talks about setting fields that are required for unique matching but the validation rules might require more fields. On the other hand, setting it as optional would allow to Observe an external resource and deleting them (using the managementPolicies: Observe, Delete) without having to fill out the spec.
+
+This is not super important for us right now since we plan to have an export cli to create the spec for us but might help to have a more common behaviour accross our providers.
+
+**TODO: how do we decide on this matter above?**
+
+Since the matching between MR and external resource is done based on external-name, a wrongly filled spec would result in Update() calls to the resource changing it. To prevent this, the user would not set the managementPolicy to Update. If there is then a missmatch, the Observe() method would determines this, sets the externalObservation.Diff and it would be visible in the debug log. 
+**TODO**: this should be put as a warning event on the resource maybe, because crossplane-runtime only does a log.Debug with the Diff. As a user adopting existing landscapes it would be good to know if the current spec WOULD trigger an update/diff for confidence.
```

Comment:
In Observe, we need to put this **Diff** somehwere: event and status.

Also see Circuit Breaker pattern from KymaEnv might be worth to put in a general concept for every resource (hash of spec fields stored in status to compare in Observe to determine if it needs update).
Circuit Breaker as follwo up.